main.js

```diff
@@ -1,64 +1,116 @@
-var page = require('webpage').create(),
-    system = require('system'),
-    fs = require('fs'),
-    zip = new require('node-zip')(),
-    _ = require('lodash-node');
+const puppeteer = require("puppeteer");
+const fs = require("fs/promises");
+const _ = require("lodash");
+const zip = new require("node-zip")();
 
-if (system.args.length != 3) {
-  console.warn("Must provide lat,lon and output directory arguments");
-  phantom.exit();
-}
+// parse arguments
+const lon = Number.parseFloat(process.argv[2]);
+const lat = Number.parseFloat(process.argv[3]);
+const colorFlag = process.argv[5];
+const sessionID = process.argv[4];
+const cookieKey = "_strava4_session";
 
-var outputFile = system.args[system.args.length - 1],
-    center = system.args[system.args.length - 2];
+const colorOptions = {
+	h: "hot",
+	g: "gray",
+	p: "purple",
+	r: "bluered",
+	b: "blue",
+};
 
-try {
-  fs.write(outputFile, '');
-}
-catch (e) {
-  console.warn('Cannot write to file ' + outputFile);
-  phantom.exit();
+if (!colorFlag) {
+	colorString = "gray";
+} else {
+	colorString = colorOptions[colorFlag] || "gray";
 }
 
-if (center.split(',').length != 2) {
-  console.warn('Must supply lat,lon in the form of lat,lon')
-  phantom.exit();
-}
-else {
-  var latlon = center.split(','),
-      lat = latlon[0],
-      lon = latlon[1];
+//check for arguments
+if (!(lon && lat && sessionID)) {
+	console.log(
+		```Must include latitude, longitude and session-id
+		Remember to use the '--' "end of options" flag to send arguments to the script
+		Optionally, you may pass a 'colorFlag'. See Readme.
+		```
+	);
+	console.log(
+		"Remember to use the '--' end of options flag to allow negative numbers"
+	);
+	process.exit(1);
 }
 
-var zoom = 15,
-    size = 1000,
-    east, west, north, south, bounds;
+//create template file
+const createTemplate = async () => {
+	const templateData = await fs.readFile("kml.template", "utf8");
+	const kmlTemplate = _.template(templateData);
+	return kmlTemplate;
+};
+
+//execution
+const reqUrl = `https://www.strava.com/heatmap#14.33/${lon}/${lat}/${colorString}/ride`;
```

Comment:
I notice in the screenshot, there is some drift between the map base layer and the heatmap image on top. Is that because the zoom level changed from 15 to 14.33 in these changes?

Reply:
It was not the zoom, but the resize event needed to be triggered after removal of dom elements. I returned to zoom #15, as its a nice integer, though.